indico/modules/events/papers/controllers/management.py

```diff
@@ -51,13 +54,27 @@ class RHManagePaperTeams(RHManagePapersBase):
     """Modify managers of the papers module"""
 
     def _process(self):
-        managers = {p.principal for p in self.event_new.acl_entries
-                    if p.has_management_role('paper_manager', explicit=True)}
-        form = PaperManagersForm(managers=managers)
+        cfp = self.event_new.cfp
+        form_data = {
+            'managers': cfp.managers,
+            'judges': cfp.judges
+        }
+        if cfp.content_reviewing_enabled:
+            form_data['content_reviewers'] = cfp.content_reviewers
+        if cfp.layout_reviewing_enabled:
+            form_data['layout_reviewers'] = cfp.layout_reviewers
+
+        form = PaperTeamsForm(event=self.event_new, **form_data)
         if form.validate_on_submit():
             update_object_principals(self.event_new, form.managers.data, role='paper_manager')
-            logger.info("Paper managers of %r updated by %r", self.event_new, session.user)
-            return jsonify_data(flash=False)
+            update_object_principals(self.event_new, form.judges.data, role='paper_judge')
+            if cfp.content_reviewing_enabled:
+                update_object_principals(self.event_new, form.content_reviewers.data, role='paper_content_reviewer')
+            if cfp.layout_reviewing_enabled:
+                update_object_principals(self.event_new, form.layout_reviewers.data, role='paper_layout_reviewer')
+            flash(_("The members of the teams were updated successfully"), 'success')
+            logger.info("Paper teams of %r updated by %r", self.event_new, session.user)
```

Comment:
Well, it's kinda unnecessary imo cause we already have the protection log entry:

![image](https://cloud.githubusercontent.com/assets/1579899/21043865/50d970ca-bdf9-11e6-9214-2290552a3ec2.png)